GitUI/CommandsDialogs/SettingsDialog/Pages/RevisionLinksExtractor.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using GitCommands.ExternalLinks;
+using ResourceManager;
+
+namespace GitUI.CommandsDialogs.SettingsDialog.Pages
+{
+    public enum CloudProviderKind
+    {
+        None = 0,
+        GitHub,
+        AzureDevOps,
+    }
+
+    public interface ICloudProviderExternalLinkDefinitionExtractor
```

Comment:
Please move each into own file

Reply:
Thanks for the early review but the refactoring is in progress...that doesn't build anymore.